### PR TITLE
test: add `features/collection` for collection related tests

### PR DESCRIPTION
**Pull request by rubiin**
Adds a separate test file for collection

### CODE REVIEW

**Comment by rubiin** [comment]
@B4nan need your input on this

File: tests/features/collection/collection.test.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+const { BookTag3 } = require('../../entities-js').BookTag3;
+
+describe('EntityManagerSqlite', () => {
+
+  let orm: MikroORM<SqliteDriver>;
+
+  beforeAll(async () => orm = await initORMSqlite());
+  beforeEach(async () => orm.schema.clearDatabase());
+
+  test('isConnected()', async () => {
```
Comment (B4nan) [inline tests/features/collection/collection.test.ts:16]: this is unrelated copypaste

File: tests/features/collection/collection.test.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    await orm.close(true);
+    expect(await orm.isConnected()).toBe(false);
+    await orm.connect();
+    expect(await orm.isConnected()).toBe(true);
+
+    // as the db lives only in memory, we need to re-create the schema after reconnection
+    await orm.schema.createSchema();
+  });
+
+
```
Comment (B4nan) [inline tests/features/collection/collection.test.ts:27]: you keep adding 2 blank lines on random places, please don't, there should be just one

File: tests/features/collection/collection.test.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    const tag2 = new BookTag3('funny');
+    const tag3 = new BookTag3('sick');
+    const tag4 = new BookTag3('strange');
+    const tag5 = new BookTag3('sexy');
+    book1.tags.add(tag1, tag3);
+    book2.tags.add(tag1, tag2, tag5);
+    book3.tags.add(tag2, tag4, tag5);
+
+    await orm.em.persist(book1);
+    await orm.em.persist(book2);
```
Comment (B4nan) [inline tests/features/collection/collection.test.ts:43]: persist is sync method

File: tests/features/collection/collection.test.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    await orm.em.persist(book3).flush();
+
+
+    const tagRepository = orm.em.getRepository<typeof BookTag3>(BookTag3);
```
Comment (B4nan) [inline tests/features/collection/collection.test.ts:47]: if you wouldn't copy this from the JS entity test, it would be cleaner. those JS entities require such weird casting when used inside TS test

File: tests/features/collection/collection.test.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    // count
+    expect(book.tags.count()).toBe(0);
+
+
+    // isEmpty
+    expect(book.tags.isEmpty()).toBe(true);
+  });
+
+
+  afterAll(async () => {
```
Comment (B4nan) [inline tests/features/collection/collection.test.ts:99]: this should go to the top ideally, where the other hooks are

**Review by B4nan — CHANGES_REQUESTED** [review]

File: tests/features/collection/collection.test.ts (hunk 5 above)
Comment (rubiin) [reply tests/features/collection/collection.test.ts:99]: Yeah I copied most from the Sqlite test file . Will do that from another file

File: tests/features/collection/collection.test.ts (hunk 2 above)
Comment (rubiin) [reply tests/features/collection/collection.test.ts]: Thats my IDE . Dont know why it happens. Perhaps we can add this rule https://eslint.org/docs/latest/rules/no-multiple-empty-lines

**Comment by B4nan** [comment]
If this should add any value, you should extract the collection tests from the EM tests - move them, so we don't just duplicate what is already tested somewhere else.

**Comment by rubiin** [comment]
> If this should add any value, you should extract the collection tests from the EM tests - move them, so we don't just duplicate what is already tested somewhere else.
> 
> And I guess I would rather do that in the v6 branch.

Yeah . I saw them on many files , one of the reason that i didnt move them out

**Comment by rubiin** [comment]
Will be closing this for now and creating a new one
